models/src/it/java/io/sphere/sdk/reviews/ReviewIntegrationTest.java

```diff
@@ -0,0 +1,392 @@
+                Review review = client().execute(new ReviewCreateCommand(ReviewDraft.of(product.getId(), customer.getId())
+                        .withAuthorName((String) REVIEWS[0][0]).withTitle((String) REVIEWS[0][1])
+                        .withText((String) REVIEWS[0][2]).withScore((Float) REVIEWS[0][3])));
+                assertNotNull(review.getId());
+            });
+        });
+
+    }
+
+    @Test
+    public void testCreate_NoAuthorName() throws Exception {
+        withProduct(client(), product ->
+                withCustomer(client(), customer -> {
+                    Review review = client().execute(new ReviewCreateCommand(ReviewDraft.of(product.getId(), customer.getId())
+                            .withTitle((String) REVIEWS[0][1])));
+                }));
+    }
+
+    @Test
+    public void testGetAll() throws Exception {
+        withProduct(client(), product -> {
+            for(int i=0; i<2; i++) {
+                withCustomer(client(), customer ->
+                    withReview(client(), product, customer, newReview -> {
+                        assertNotNull(newReview.getId());
+                        PagedQueryResult<Review> result = null;
+                        boolean notFound = true;
+                        int offset = 0;
+                        while((result==null || !result.isLast()) && notFound) {
+                            result = client().execute(ReviewQuery.of().withOffset(offset));
+                            Optional<Review> optReview =
+                                    result.getResults().stream().filter(review1 -> review1.getId().equals(newReview.getId())).findFirst();
+                            notFound = !optReview.isPresent();
+                            offset = result.getOffset()+result.getResults().size();
+                        }
+                        assertFalse(notFound);
+                    }));
+            }
+        });
+    }
+
+    @Test
+    public void testGetOne() throws Exception {
+        withProduct(client(), product ->
+                withCustomer(client(), customer ->
+                        withReview(client(), product, customer, review -> {
+                            PagedQueryResult<Review> result = client().execute(ReviewQuery.of()
+                                    .withPredicate(Predicate.of("id = \"" + review.getId() + "\"")));
+                            Optional<Review> optReview =
+                                    result.getResults().stream().filter(review1 -> review1.getId().equals(review.getId())).findFirst();
+                            assertTrue(optReview.isPresent());
+                            assertEquals(optReview.get(), review);
+                        })));
+    }
+
+    @Test
+    public void testGetById() throws Exception {
+        withProduct(client(), product ->
+                withCustomer(client(), customer ->
+                        withReview(client(), product, customer, review -> {
```

Comment:
The block

```
        withProduct(client(), product ->
                withCustomer(client(), customer ->
                        withReview(client(), product, customer, review -> {
```
Is repeated a lot, I think it would be nicer if "withReview" takes care of creating a product a customer underneath.